backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/participation/ParticipationController.java

```diff
@@ -30,15 +30,17 @@ public List<ParticipationDto> getVolunteerParticipations(Principal principal) {
     }
 
     @PostMapping("/activities/{activityId}/participations")
-    @PreAuthorize("(hasRole('ROLE_MEMBER')) and hasPermission(#activityId, 'ACTIVITY.MEMBER')")
-    public ParticipationDto createParticipation(@PathVariable Integer activityId, @Valid @RequestBody ParticipationDto participationDto) {
-        return participationService.createParticipation(activityId, participationDto);
+    @PreAuthorize("(hasRole('ROLE_MEMBER') and hasPermission(#activityId, 'ACTIVITY.MEMBER')) or (hasRole('ROLE_VOLUNTEER') and hasPermission(#activityId, 'PARTICIPATION.CREATOR'))")
+    public ParticipationDto createParticipation(@PathVariable Integer activityId, @Valid @RequestBody ParticipationDto participationDto, Principal principal) {
+        int userId = ((AuthUser) ((Authentication) principal).getPrincipal()).getUser().getId();
```

Comment:
I would suggest that it is not necessary to keep the info on the member that created the participation